products/accredited-representation-management/technical-documentation/appoint-a-representative/poa_request_proposal.md

```diff
@@ -0,0 +1,160 @@
+# Storage Proposal for Power of Attorney Requests
+This proposal only covers the storing of data for Power of Attorney Requests.
+
+## Background
+The Office of the Chief Technology Officer (OCTO) is seeking to migrate Veteran and Power of Attorney (PoA) experiences from the legacy Stakeholder Enterprise Platform (SEP) product and eBenefits platform to VA.gov.
+
+One aspect of this experience is the Power of Attorney (POA) Request, a data object that exists to enable accredited representatives accepting or denying requests to act as a Power of Attorney (Forms 21-22 and 21-22a) on behalf of Veterans. The data object bridges the gap from the Veteran facing form fill experience (Appoint a Representative) to establishing the Power of Attorney relationship in systems of record and updating representative permissions in those systems (Lighthouse auto-establishment APIs). [See the flow chart](#data-flow)
+
+## Problem
+In order to create a functional user interface for accredited representatives ([see the sample UI](#sample-ui)), the POA Request objects must support reasonably efficient searching or filtering on the following Veteran attributes: first name, last name, city, state, zip code, and consent limits. The front end must also be able to display information related to change of address authorization and health record access limitations.
+
+There is not an existing pattern or solution in vets-api for storing Veteran PII that is secure and efficiently searchable for a many row result.
+
+## Solution
+Code below is for illustrative purposes. These snippets are not likely to be the final implementation.
+
+### Column level encryption
+The [blind index gem](https://github.com/ankane/blind_index) allows for exact matching queries on encrypted columns. This will be used on fields that users will search or filter by. Existing KMS encryption will be used on the fields that are for display purposes only.
+
+#### The Model
+```ruby
+class PowerOfAttorneyRequest < ApplicationRecord
+  has_kms_key
+  has_encrypted :consent_data, key: :kms_key, **lockbox_options
```

Comment:
I think that consent data is a particularly important data point for a given POA request. We have heard that some VSOs intend to reject any request that don't consent to everything. There is a chance that product would like users to be able to filter by this data point.

Reply:
There were a few things like this that were tough to make a call on without a final product decision. Maybe a good compromise would be to keep the consent data jsonb for the specific limitations and make a boolean `has_health_record_limitations` that is `true` if any of the 4 limitations are present at record creation.

Reply:
Yeah that makes sense about product ambiguity. And you've called out that this document is meant to be sufficiently illustrative about security concerns. It's possible that these data points correctly fit within that idea of fields (possibly in their own table) that make the POA request non-inert / searchable / actionable and can get the blind-index + expiration treatment.